internal/charmstore/migrations.go

```diff
@@ -222,3 +227,96 @@ func populateWriteACL(db StoreDatabase) error {
 	logger.Infof("%d base entities updated", counter)
 	return nil
 }
+
+func populatePromulgatedEntities(db StoreDatabase) error {
+	baseEntities := db.BaseEntities()
+	entities := db.Entities()
+
+	// 1. Update All Base Entities to have a promulgated value of the correct form.
+	_, err := baseEntities.UpdateAll(
+		bson.D{{"promulgated", true}},
+		bson.D{{"$set", bson.D{{"promulgated", 1}}}},
+	)
+	if err != nil {
+		return errgo.Notef(err, "cannot set promulgated to initial value")
+	}
+	_, err = baseEntities.UpdateAll(
+		bson.D{{"$or", []interface{}{
+			bson.D{{"promulgated", false}},
+			bson.D{{"promulgated", bson.D{{"$exists", false}}}},
+		}}},
+		bson.D{{"$set", bson.D{{"promulgated", -1}}}},
+	)
+	if err != nil {
+		return errgo.Notef(err, "cannot set promulgated to initial value")
+	}
+
+	// 2. Update entities with users with their promulgated URL.
```

Comment:
// Note that we order the query results by revision, so when we've traversed all the entities,
// owner[name] will hold the owner of the latest revision of each promulgated
// charm with the given name.

?